### PR TITLE
[Themes] Remove Ruby from `theme push` command

### PR DESCRIPTION
**Pull request by jamesmengo**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/develop-advanced-edits/issues/352

### WHAT is this pull request doing?
- Removes 
- Removes Ruby invocation for the `theme push` command

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/theme/src/cli/commands/theme/push.ts [hunk 1]
```diff
@@ -135,7 +130,7 @@ export default class Push extends ThemeCommand {
   async run(): Promise<void> {
     const {flags} = await this.parse(Push)
 
-    if (flags.password || flags.legacy) {
+    if (flags.password) {
```
Comment (jamesmengo) [inline packages/theme/src/cli/commands/theme/push.ts:133]: Blocked by: https://github.com/orgs/Shopify/projects/4613/views/62?pane=issue&itemId=80272317

**Comment by 50bbx** [comment]
As I can see from the [docs](https://shopify.dev/docs/api/shopify-cli) the ruby dependency is not required anymore but this changes are still in draft. Was that intended?

And it looks like ruby is actually still required (version 3.67.0): image

Is this true?

**Comment by jamesmengo** [comment]
@50bbx You should be able to run the TS implementation now without the need to install Ruby (depending on the version)

Could you share which version you're on + which command you're running? The `--legacy` or `password` flag may be configured in your environment variables, which is causing it to default to Ruby.

**Comment by jamesmengo** [comment]
npm snapshot

File: packages/theme/src/cli/commands/theme/push.ts [hunk 2]
```diff
@@ -135,11 +121,6 @@ export default class Push extends ThemeCommand {
   async run(): Promise<void> {
     const {flags} = await this.parse(Push)
 
-    if (flags.password || flags.legacy) {
```
Comment (jamesmengo) [inline packages/theme/src/cli/commands/theme/push.ts:138]: [Validate \`Push\` command output in CI/CD workflows · Issue #337 · Shopify/develop-advanced-edits](https://github.com/Shopify/develop-advanced-edits/issues/337)

**Review by frandiox — APPROVED** [review]
#feelsgood

**Review by karreiro — APPROVED** [review]
Thank you, @jamesmengo! LGTM and works as expected! I believe we just need to remove the `cli2Flags` variable, and then we will be ready to merge :)

**Pull request merged**